docs/developer/OPERATIONS.md

```diff
@@ -44,30 +57,65 @@ git checkout -b release-X.Y.Z-prep
 scripts/update_release.sh X.Y.Z
 ```
 
-This script will also change some things in the operations repository, those should
-also be submitted via a PR to the appropriate repository and merged after tag creation.
-
-Check and eventually update [CHANGELOG](../CHANGELOG.md).
+Check and eventually update [CHANGELOG](../CHANGELOG.md). If everything looks OK
 
 ```bash
 git commit -am 'Update version to X.Y.Z'
 git push -u origin release-X.Y.Z-prep
 ```
 
-And then submit `release-X.Y.Z-prep` as a PR against the `release-X.Y.Z` branch.
+and submit `release-X.Y.Z-prep` as a PR against the `release-X.Y.Z` branch.
+After the review is successfully finished and the PR is merged, the release branch is ready.
+You can continue with [UAT](###UAT) phase.
+
+### UAT
+
+Deploy the `release-X.Y.Z` branch to the **UAT** environment - the details on
+the deployment procedure can be found in the OPS repository documentation.
+Announce the upcoming release to the users and ask them to test their
+workflows or integrations using the **UAT** environment.
+
+> [!IMPORTANT]
+> In the case of major release it is necessary to namely announce all the breaking
+> changes and expressly ask the owners of all the known integrations to account for
+> them to prevent any posible breakage. It may happen that some of them will
+> require an extended adoption period especially if the changes are significant.
+
+Gather the feedback over time and fix the reported bugs with priority.
+This pre-release testing period should last at least one week or as long as needed
+to fix all the newly reported bugs not to introduce any regression.
+
+> [!NOTE]
+> At this point only bug fixes should be added into the release branch.
+> Use `git cherry-pick` to keep it in sync with `master`.
+
+### Production
+
+After the successful [UAT](###UAT) phase, it is time for the production release.
+Put together the release announcement in advance so it can be sent right after the
+release. Announce to the users that there may be a short period of production
```

Comment:
Should we mention where this is sent to? OSIDB-announce list?

Reply:
do we also want to add a link to announcement drafts? https://docs.google.com/document/d/1w6Mv_CVyvcG5MVQjsnK02YuBO_axSlaKXIwXUwV90pQ/edit#heading=h.pdqq138mt631

Reply:
@jvaught-rh @znovacek the details on this are not added intentionally as it is an internal information and the repository is public + aiming to be potentially a useful project for not only RH and this information would not be useful for anyone else. If we want to document it then elsewhere.